scripts/build_native_deps.sh

```diff
@@ -440,10 +408,28 @@ build_python()
         "$src/configure" --prefix="$PYTHON_INSTALL" --disable-shared --enable-static
         $MAKE -j$PROCS
         $MAKE install
-    ) > "$LOG/python.log" 2>&1
+    ) > "$LOG/python.log" 2>&1 || print_failed_and_exit "$LOG/python.log"
 )
+
+
+# Run build functions.
+build_expat
+build_freetype2
+build_util_linux
+build_fontconfig
+build_ghostscript
+build_libffi
+build_zlib
+build_glib2
+build_gmp
+build_libtool
+build_guile
+build_pixman
+build_cairo
+build_harfbuzz
+build_pango
```

Comment:
I still don't understand why you want to move the invocation away from the definition (and right now don't have an `echo` between them). Maybe to give some context, putting the commands into the functions is needed because I don't want variables to leak into global space. So think of them as a code block with local variables.

If there's an advantage to have all calls at the end of the file, I'm totally fine with the previous loop (if that is portable). Maybe you can briefly comment on your reasoning here?

Reply:
Ah, I _do_ remember why I collected them in an array and looped over them:

- It made it easy to quickly comment out one or two of them while testing.
- Instead of a lot of explicit `echo`s, we only need one (in the loop.)
- IMO the script is more readable when function definitions and function calls live in separate blocks.